### ISSUE TITLE
Oral histories citations include incorrect n.d. reference when date is present

### ISSUE THREAD

**Issue by yonyitz** [sciencehistory/scihist_digicoll#574]
N.d. is erroneously present after Science History Institute in all oral history records. 

See example here: "Fred Basolo, interviewed by James J. Bohning in Northwestern University (Evanston, Ill.) on March 1, 1991. Philadelphia: Science History Institute, n.d. Oral History Transcript 0091."

Please remove n.d. in citation unless there is a reason why it is present.

**Comment by eddierubeiz** [comment]
@sanfordd  notes:
"What's occurring is that the date of interview is being used in the title section when they say "on March 1, 1991" and then it's been set to a nil value which the CSL (Citation Style Language) tool we're using reads as no date and inserts as `n.d.`in the date area."

**Comment by jrochkind** [comment]
Hmm, I wonder if a new release of the ruby CSL gem  changed behavior there. 

We could fairly easily make the date appear _twice_. 

> "Fred Basolo, interviewed by James J. Bohning in Northwestern University (Evanston, Ill.) on March 1, 1991. Philadelphia: Science History Institute, March 1, 1991.  Oral History Transcript 0091."

Or something like that. But that's not what you're asking for. 

We may not very easily be able to make the date not appear at all in that second slot.  The CSL style for "chicago style" that we are using believes there must be a date at that position in a chicago-style citation. Either an actual date, or "n.d." for no-date. 

We could look into defining our own custom CSL style instead of using the built-in "chicago", that has different behavior. 

It might also be worth looking at old chf_sufia app to see if it behaves identically. If it behaves differently and does not put the "n.d." there, then looking into how the code may differ may give us a clue as to a solution.

**Comment by eddierubeiz** [comment]
For published oral histories, there's a distinction to be made (if we want to) between the **interview** date and the **publication** date. Ideally, the **interview** date would go in the title, and the **publication** date would go where it currently says `n.d.`.

**Comment by jrochkind** [comment]
I don't believe we have the publication date recorded in our metadata, or ever have. 

Looks like we did do some custom work on "cite as" for Oral Histories, at https://github.com/sciencehistory/chf-sufia/issues/1169

Still curious what it was looking like in chf_sufia; not sure if it's been this way since we "improved" it, and we didn't notice the problem (or thought it was okay at the time); or if it somehow got broken in between chf_sufia days and this app. The main reason to be curious is if it *was* working better in chf_sufia, that will give us a path to make it work better again. If it never was, there may no good way to get it to.

**Comment by yonyitz** [comment]
Really great point about the difference between the publisher date and interview date.  Also some records have multiple interview dates.  Not sure how to handle that...

**Comment by eddierubeiz** [comment]
Ashley she says the library needs a "publication date" to manufacture the call numbers affixed to the bound copies of the oral history transcripts. They just use the interview date for that purpose.

**Comment by yonyitz** [comment]
Cool.  What if there are multiple interview dates?

**Comment by eddierubeiz** [comment]
They use the first.

**Comment by yonyitz** [comment]
Ok.  That makes sense!  Let me know if I need to do anything on my end!

**Comment by HKativa** [comment]
I'm on board with following the library's example and using the first interview date as the "publication" date for the purposes of citation. Also, having the date appear twice in the citation, per Jonathan's example at the top of the thread, works for me. As it stands, the issue isn't with the inclusion of "n.d." or another date in the citation, but rather that there exists a date we COULD input there rather than using "n.d."

**Comment by jrochkind** [comment]
As I discovered finding the issue on chf_sufia, what happened here is we were trying to make the citation look more like it looks in the "cite as" section of the OPAC for these records. (Of course, that wasn't necessarily "Chicago" style, but it was close). 

Part of that was putting "on DATE" at the end of the title in the citation, even though it wasn't in our title metadata. 

We thought we could have it *only* there, but at least in the current app with present versions of CSL libraries, that results in the `n.d.`. 

So, we aren't accomplishing what we meant to. 

We could have the date "in two places". We could *instead*  just *stop* trying to put it in the title as "on DATE" to match the OPAC-suggested citation, and have it only at the end in the proper Chicago slot. Is the latter preferable @HKativa ?

**Comment by HKativa** [comment]
@jrochkind thanks for the extra context. Are the options outlined above roughly equivalent in terms of developer time?

**Comment by jrochkind** [comment]
Yep, now that we've figured out what was going on (to some extent), it ought to be relatively easy to make either of those changes.

**Comment by jrochkind** [comment]
Just for my own curiosity... yes, the "n.d." has been there since we did the work to improve OH citations ~1 year ago, which can be verified cause a test was included at the time for what we expected citation to look like, and "n.d." was there, ok! So OH citations have looked this way for a year, the app migration did not change it. 

https://github.com/sciencehistory/chf-sufia/pull/1214/files#diff-e401084378064cd527614519b0825b75R393
